agent/src/main/scala/io/scalac/mesmer/agent/akka/dispatcher/ActorCellConstructorAdvice.scala

```diff
@@ -0,0 +1,49 @@
+package io.scalac.mesmer.agent.akka.dispatcher
+
+import akka.actor.typed.scaladsl.adapter.ClassicActorSystemOps
+import akka.dispatch.Dispatcher
+import akka.dispatch.MessageDispatcher
+import net.bytebuddy.asm.Advice.Argument
+import net.bytebuddy.asm.Advice.OnMethodEnter
+
+import io.scalac.mesmer.core.event.DispatcherEvent.ExecutorConfigEvent
+import io.scalac.mesmer.core.event.EventBus
+
+object ActorCellConstructorAdvice {
+
+  @OnMethodEnter
+  def constructor(
+    @Argument(0) system: akka.actor.ExtendedActorSystem,
+    @Argument(3) dispatcher: MessageDispatcher
+  ): Unit =
+    if (dispatcher.id == "akka.actor.default-dispatcher") {
+      DispatcherDecorator.setActorSystem(dispatcher.asInstanceOf[Dispatcher], system)
+
+      val akkaConfig       = system.settings.config.getConfig("akka.actor")
+      val dispatcherConfig = akkaConfig.getConfig("default-dispatcher")
+      dispatcherConfig.getString("executor") match {
```

Comment:
I still wonder about these _static_ props... If we always take these from the config and they never change, does it make sense to make a Mesmer feature out of it at all? After all users can reach their configs. Unless there are scenarios when these properties can change but I'm not aware?

@ornel-lloyd-edano @worekleszczy what do you think?

Reply:
Also, notice that other solutions (Kamon) do not rely on the config (at least directly). Not sure why (I would need to investigate myself).